enhancements/ingress/mutable-publishing-scope.md

```diff
@@ -0,0 +1,348 @@
+the default IngressController as well as custom IngressControllers.
+
+### Goals
+
+1. The cluster administrator can change the scope of the service load-balancer of an IngressController with "LoadBalancerService" endpoint publishing strategy type.
+2. If the cloud platform supports mutating scope on a service load-balancer, the operator will perform this operation when the cluster administrator changes the configured scope.
+3. If the cloud platform does not support mutating scope, the operator will advise the user how to complete the operation.
+
+### Non-Goals
+
+1. The operator does not delete a service load-balancer for an IngressController that is not marked for deletion.
+
+## Proposal
+
+Some platforms (such as Azure and GCP) support changing the scope of a service
+load-balancer between internal and external without deleting and recreating the
+load balancer, by setting an annotation on the Kubernetes Service object.  On
+these platforms, the operator merely sets the annotation to the desired scope,
+and the operation of changing it scope is complete.
+
+Other platforms (such as AWS) require deleting and recreating a load balancer to
+change its scope.  This operation is disruptive: It interrupts ingress traffic
+and may cause the load balancer's address to change.  On these platforms, the
+operator signals that the user must delete the Kubernetes Service object.  Once
+the user performs this step, the operator recreates the load balancer with the
+desired scope to complete the operation.
+
+In order to signal that the user must take action, the operator sets the
+`Progressing` status condition to `True` with a message that indicates the
+action that the user must take:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: custom
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  domain: mycluster.com
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: Internal
+status:
+  availableReplicas: 2
+  conditions:
+  - message: Deployment has minimum availability.
+    reason: MinimumReplicasAvailable
+    status: "True"
+    type: Available
+  - message: Have load balancer with scope "External", want load balancer with scope "Internal".  You can delete the openshift-ingress/router-default service to proceed [...].  Alternatively, you can change the IngressController's spec.endpointPublishingStrategy.loadBalancer.scope field value back to its previous value [...].
+    reason: ScopeChanged
+    status: "True"
+    type: Progressing
+```
+
+Note that the IngressController remains available, and that the user can revert
+the change by setting `spec.endpointPublishingStrategy.loadBalancer.scope` back
+to its previous value.  That means the user must take one of two actions:
+
+1. Delete the Service referenced in the status condition.
+2. Revert the change to the IngressController.
```

Comment:
When we introduced an earlier version of this feature (which we since reverted) that automatically deleted and recreated the service, we had at least one user who changed the scope and then realized it would require deprovisioning and recreating the LB and wanted to revert the change.  The option to back out the change is closely tied to the motivation for this enhancement; I'll revise the motivation section to call this out.

Reply:
Ok